src/python/pants/backend/python/goals/tailor.py

```diff
@@ -66,6 +96,62 @@ async def find_putative_targets(
                     tgt_type, dirname, name, sorted(filenames), kwargs=kwargs
                 )
             )
+
+    if python_setup.tailor_pex_binary_targets:
+        # Find binary targets.
+
+        # Get all files whose content indicates that they are entry points.
+        digest_contents = await Get(DigestContents, PathGlobs, all_py_files_globs)
```

Comment:
This is slightly unfortunate since we've already had queries about supporting python scripts w/o a `.py` extension. Handling those could maybe come later. I'd guess they'd have a revealing shebang to help identify further.

Reply:
Yeah, I think it's OK to not handle every case. If you have such a script, you'd have to add it manually to a `python_library` anyway, so adding the `pex_binary` as well doesn't seem like the worst thing.